torch/distributed/_spmd/api.py

```diff
@@ -426,7 +426,7 @@ class _CompiledResult:
 
 def _compile(
     func: Callable,
-    module_override: Optional[Dict[Type[Any], Override]],
+    module_override: Optional[Dict[Union[Type[Any], str], Override]],
```

Comment:
shall we only support `fqn` based override instead of module key based one if we are only going to use the `fqn` based override?

Reply:
not sure yet, we hit similar problem with composable APIs where there might be multiple submodule instances of the same type but with different names. In such case, type-based seems more user friendly, but FQN-based key is indeed more flexible.